pkg/util/proto/testing/openapi.go

```diff
@@ -52,6 +56,35 @@ func (f *Fake) OpenAPISchema() (*openapi_v2.Document, error) {
 	return f.document, f.err
 }
 
+func (f *Fake) OpenAPIV3Discovery() (*handler3.OpenAPIV3Discovery, error) {
+	// Read directory to determine groups
+	res := &handler3.OpenAPIV3Discovery{}
+	filepath.WalkDir(f.Path, func(path string, d fs.DirEntry, err error) error {
+		if filepath.Ext(path) != "json" {
+			return nil
+		}
+
+		res.Paths[path] = handler3.OpenAPIV3DiscoveryGroupVersion{
+			URL: "/openapi/v3/" + path,
+		}
+		return nil
+	})
+
+	return res, nil
+}
+
+func (f *Fake) OpenAPIV3Schema(groupVersion string) (*openapi_v3.Document, error) {
+	_, err := os.Stat(f.Path)
+	if err != nil {
+		return nil, err
+	}
+	spec, err := ioutil.ReadFile(filepath.Join(f.Path, groupVersion+".json"))
+	if err != nil {
+		return nil, err
+	}
+	return openapi_v3.ParseDocument(spec)
+}
```

Comment:
This one doesn't cache the result, which is sort of the promise of the `Fake` object, maybe we need a `FakeV3`?

Reply:
I added caching to this method of the `Fake`.

On a slightly related note,  I did end up adding the following mock class to `client-go` which I found more useful in my testing:

https://github.com/alexzielenski/kubernetes/blob/cache-busting/staging/src/k8s.io/client-go/util/testing/fake_openapi_handler.go#L40

Maybe it is more appropriate to add to kube-openapi, but I did not use it there. I use this mock server so far in tests for `client-go`, `kubectl`, and `kube-aggregator`